src/sagas/login.js

```diff
@@ -234,6 +236,15 @@ function* initialLoad () {
   yield put(getByPage({ page: 1 }))
 }
 
+function* fetchWebsocketInformation () {
+  const data = yield callExternalAction('getOption', Api.getOption, getOption('WebSocketProxy', 'general', ''))
```

Comment:
Do we know if it is possible to connect NoVNC if the websocket proxy isn't available?

If websocket proxy is required, do we have handling on the web console connection if it is not defined?  (I haven't read through it in enough detail to remember at the moment)

Reply:
> Do we know if it is possible to connect NoVNC if the websocket proxy isn't available?

Generally `WebSocketProxy` is a URL through which client will connect to NoVNC console, if it is unavailable, then client can't connect to NoVNC.

> If websocket proxy is required, do we have handling on the web console connection if it is not defined? (I haven't read through it in enough detail to remember at the moment)

Only what I found about it: `If the console does not start up automatically, this empty state will appear. The user can click the primary or secondary ‘Connect’ button to start up the console.`

I think handling of this situation can be moved to separate issue, because this PR is already too big, so it is better to split next issues.

Reply:
Detecting a missing or empty websocket proxy url (or a simple sanity test to make sure it looks right) followed by an error message or a toast would probably be enough.  It is related to if the proxy is down, or if the url has an error, or if the engine root cert isn't approved on the browser and wss:// is blocked.  They're all instance of the websocket not being right and that the connection will fail because of that.

@lwrigh should weigh in on what kind of error display/response is reasonable.  Toast, error message box?  Distinct messages for each kind of problem or a generic "can't connect to your VM's console due to proxy errors" message?  I think hiding the browser console button would be confusing.  @sgratch (or even @gregsheremeta) - Thoughts?

@leistnerova made a good point at standup that with little time pressure currently, handling this edge case in this PR is probably worth it (even if it is super annoying).

Reply:
I would definitely avoid hiding the browser console button, that would make for a really confusing experience. I think in this case it would make the most sense to feature the message dialog component (https://www.patternfly.org/pattern-library/communication/message-dialog/#design) and feature a message about the proxy error. Would like this make sense as an error message,"Cannot connect to [Name of VM]'s console because of a proxy error. Please try again.". ? The actions on the dialog could be 'Cancel' and 'Try Again'.